io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -516,22 +527,32 @@ newtype TVarId    = TVarId    Int deriving (Eq, Ord, Enum, Show)
 newtype TimeoutId = TimeoutId Int deriving (Eq, Ord, Enum, Show)
 newtype ClockId   = ClockId   Int deriving (Eq, Ord, Enum, Show)
 
+unTimeoutId :: TimeoutId -> Int
+unTimeoutId (TimeoutId a) = a
+
 type ThreadLabel = String
+type TVarLabel   = String
 
-data LabelledThread = LabelledThread {
-    labelledThreadId    :: ThreadId,
-    labelledThreadLabel :: Maybe ThreadLabel
+data Labelled a = Labelled {
+    l_labelled :: !a,
+    l_label    :: !(Maybe String)
   }
   deriving (Eq, Ord, Generic)
-  deriving Show via Quiet LabelledThread
+  deriving Show via Quiet (Labelled a)
+
+type LabelledThreadId = Labelled ThreadId
+type LabelledTVarId = Labelled TVarId
```

Comment:
Do we need these type aliases? Isn't it clearer without? It's also only a single ' ' difference :-)

Reply:
I'll remove them.